src/core/note_app.dart

```diff
@@ -1,6 +1,136 @@
+import 'dart:io';
+import '../models/note_model.dart';
+
 class NoteApp {
-  // TODO: Complete the run function
+  List<NoteModel> notes = [];
+
   void run() {
-    print('Hello, world!');
+    while (true) {
+      displayMenu();
+
+      stdout.write('Enter your choice (1-5): ');
+      final choice = int.tryParse(stdin.readLineSync() ?? '');
+
+      if (choice == null || choice < 1 || choice > 5) {
+        print('Invalid choice. Please try again.');
+        continue;
+      }
+
+      switch (choice) {
+        case 1:
+          createNote();
+          break;
+        case 2:
+          editNote();
+          break;
+        case 3:
+          deleteNote();
+          break;
+        case 4:
+          searchNote();
+          break;
+        case 5:
+          print('Exiting...');
+          return;
+      }
+    }
   }
-}
+
+  void createNote() {
+    stdout.write('Enter note title: ');
+    final title = stdin.readLineSync() ?? '';
```

Comment:
You should check that the user hasn't entered a null title, as you don't want the title to be without a name.